src/compiler/utilities.ts

```diff
@@ -8205,7 +8205,7 @@ export function setObjectAllocator(alloc: ObjectAllocator) {
 
 /** @internal */
 export function formatStringFromArgs(text: string, args: DiagnosticArguments): string {
-    return text.replace(/{(\d+)}/g, (_match, index: string) => "" + Debug.checkDefined(args[+index]));
+    return text.replace(/\{(\d+)\}/g, (_match, index: string) => "" + Debug.checkDefined(args[+index]));

```

Comment:
Why does this have to change? Isn't this legal (but suspicious) syntax?

Reply:
Yes, this special case is currently not handled because I consider it a bad practice to not escape the character despite being in non-Unicode mode, just like what we are doing so now for octal escape.
But as I just commented I understand that this case seems to be common in codebases. Yet, I don’t want to totally silence the error, since if someone writes /a{1,3/ it’s very likely that the user forgets the closing brace.
So, a compromise would be to error if there are digits directly following the opening brace. Does it sound right?

Reply:
I feel like there are lint rules which are intended to catch this...

Reply:
(Indeed, and this defeats the whole point of working on it…)